pkg/middleware/headers.go

```diff
@@ -0,0 +1,40 @@
+package middleware
+
+import (
+	"context"
+	"net/http"
+)
+
+type contextKey int
+
+const headerKey contextKey = 0
+
+func NewProxyHeaders(h http.Handler, headers []string) *ProxyHeaders {
+	return &ProxyHeaders{
+		h:       h,
+		headers: headers,
+	}
+}
+
+type ProxyHeaders struct {
+	h       http.Handler
+	headers []string
+}
+
+func (p *ProxyHeaders) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	hdrs := make(map[string]string, len(p.headers))
+	for _, header := range p.headers {
+		hdrs[header] = r.Header.Get("Authorization")
```

Comment:
Should "Authorization" be the value of the header variable, as we're copying the configured headers.

Maybe a check should be added to make sure .Get() does not return an empty string (which would be the case if the header does not exist in the request), and then only add the header to hdrs if the header value exists in the request.

I might also be tempted to make hdrs a map[string][]string (or http.Header) (as a header can have multiple values), and then call r.Header.Values() to return all of the values for the configured header.

Reply:
I made these changes locally, and tried it out. The configured headers are being passed through to Prometheus now. I can submit a PR if you agree with these changes.

Reply:
I fixed the hard-coded header name (thanks!); I also added the check for empty value.

I'm not sure how we want to handle multi-value headers TBH; do we want to merge? Do we want to overwrite? its a bit unclear to me. So I'm thinking for now we start with single value -- and we can update if a mutli-value use-case shows up.

Reply:
Thanks @jacksontj. Single value works for my use case.

I see the code does not compile since the last commit. Maybe map[string][]string should be changed back to map[string]string if only single values will be supported for now.